sklearn/metrics/cluster/tests/test_supervised.py

```diff
@@ -370,3 +379,70 @@ def test_check_clustering_error():
 
     with pytest.warns(UserWarning, match=msg):
         check_clusterings(wavelength, noise)
+
+
+def test_pair_confusion_matrix():
+    n = 10
+    N = n * n
+
+    # edge case: every element is its own cluster
+    clustering1 = list(range(N))
+    clustering2 = clustering1
+    expected = np.array([[N * (N - 1), 0], [0, 0]])
+    assert_array_equal(expected, pair_confusion_matrix(clustering1,
+                                                       clustering2))
+
+    # edge case: only one cluster
+    clustering1 = np.full((N,), fill_value=0)
+    clustering2 = clustering1
+    expected = np.array([[0, 0], [0, N * (N - 1)]])
+    assert_array_equal(expected, pair_confusion_matrix(clustering1,
+                                                       clustering2))
+
+    # regular case: different non-trivial clusterings
+    clustering1 = np.array([i+1 for i in range(n) for j in range(n)])
+    clustering2 = np.array([i+1 for i in range(n) for j in range(n+1)][:N])
+    # basic quadratic implementation
+    expected = np.full(shape=(2, 2), fill_value=0, dtype=np.int64)
+    for i in range(len(clustering1)):
+        for j in range(len(clustering2)):
+            if i != j:
+                same_cluster_1 = int(clustering1[i] == clustering1[j])
+                same_cluster_2 = int(clustering2[i] == clustering2[j])
+                expected[same_cluster_1, same_cluster_2] += 1
+    assert_array_equal(expected, pair_confusion_matrix(clustering1,
+                                                       clustering2))
+
+
+def test_rand_score():
+    n = 10
+    N = n * n
+
+    # edge case: every element is its own cluster
```

Comment:
We could have separated tests for normal case and edge case. I think that we can parametrize the test:

```python
@pytest.mark.parametrize(
    "clustering1, clustering2",
    [(range(100), range(100)),
     (np.zeros((100,)), np.zeros((100,))]
)
def test_rand_score_edge_case(clustering1, clustering2):
    assert_array_almost_equal(
        1.0, rand_score(clustering1, clustering2)
    )
```

Reply:
we could have something similar for one the edges cases above

Reply:
Done.